openvasd/openvasd.h

```diff
@@ -0,0 +1,312 @@
+ * @brief API for Openvas Daemon communication.
+ */
+
+#ifndef _GVM_OPENVASD_H
+#define _GVM_OPENVASD_H
+
+#include "../base/nvti.h"
+#include "../util/jsonpull.h"
+
+#include <glib.h>
+#include <stdio.h>
+#include <time.h>
+
+/** @brief Struct to hold an scan result */
+struct openvasd_result
+{
+  unsigned long id;
+  gchar *type;
+  gchar *ip_address;
+  gchar *hostname;
+  gchar *oid;
+  int port;
+  gchar *protocol;
+  gchar *message;
+  gchar *detail_name;
+  gchar *detail_value;
+  gchar *detail_source_type;
+  gchar *detail_source_name;
+  gchar *detail_source_description;
+};
+
+/** @brief Openvasd Errors */
+enum OPENVASD_ERROR
+{
+  OPENVASD_INVALID_OPT,
+  OPENVASD_NOT_INITIALIZED,
+  OPENVASD_INVALID_VALUE,
+  OPENVASD_ERROR,
+  OPENVASD_OK,
+};
+
+/** @brief Openvasd options for the connector builder */
+enum OPENVASD_CONNECTOR_OPTS
+{
+  OPENVASD_CA_CERT,
+  OPENVASD_CERT,
+  OPENVASD_KEY,
+  OPENVASD_API_KEY,
+  OPENVASD_SERVER,
+  OPENVASD_HOST,
+  OPENVASD_SCAN_ID,
+  OPENVASD_PORT,
+};
+
+enum OPENVASD_RESULT_MEMBER_STRING
+{
+  TYPE,
+  IP_ADDRESS,
+  HOSTNAME,
+  OID,
+  PROTOCOL,
+  MESSAGE,
+  DETAIL_NAME,
+  DETAIL_VALUE,
+  DETAIL_SOURCE_NAME,
+  DETAIL_SOURCE_TYPE,
+  DETAIL_SOURCE_DESCRIPTION,
+};
+
+enum OPENVASD_RESULT_MEMBER_INT
+{
+  ID,
+  PORT,
+};
+
+/**
+ * @brief Openvasd scan status.
+ */
+typedef enum
+{
+  OPENVASD_SCAN_STATUS_ERROR = -2,  /**< Error status. */
+  OPENVASD_SCAN_STATUS_FAILED = -1, /**< Failed status. */
+  OPENVASD_SCAN_STATUS_STORED,      /**< Stored status */
+  OPENVASD_SCAN_STATUS_REQUESTED,   /**< Queued status */
+  OPENVASD_SCAN_STATUS_RUNNING,     /**< Running status. */
+  OPENVASD_SCAN_STATUS_STOPPED,     /**< Stopped status. */
+  OPENVASD_SCAN_STATUS_SUCCEEDED,   /**< Succeeded status */
+} openvasd_status_t;
+
+struct openvasd_response
+{
+  long code;     /**< HTTP code response. */
+  gchar *body;   /**< String containing the body response. */
+  gchar *header; /**< A header value. */
+};
+
+struct openvasd_scan_status
+{
+  time_t start_time;
+  time_t end_time;
+  int progress;
+  openvasd_status_t status;
+  long response_code;
+};
+
+typedef struct openvasd_response *openvasd_resp_t;
+
+typedef enum OPENVASD_RESULT_MEMBER_INT openvasd_result_member_int_t;
+
+typedef enum OPENVASD_RESULT_MEMBER_STRING openvasd_result_member_string_t;
+
+typedef enum OPENVASD_CONNECTOR_OPTS openvasd_conn_opt_t;
+
+typedef enum OPENVASD_ERROR openvasd_error_t;
+
+typedef struct openvasd_result *openvasd_result_t;
+
+typedef struct openvasd_connector *openvasd_connector_t;
+
+typedef struct openvasd_scan_status *openvasd_scan_status_t;
+
+// Functions to build/free request data
+openvasd_connector_t
+openvasd_connector_new (void);
+
+openvasd_error_t
+openvasd_connector_builder (openvasd_connector_t *, openvasd_conn_opt_t,
+                            const void *);
+
+openvasd_error_t
+openvasd_connector_free (openvasd_connector_t *);
+
+void openvasd_response_free (openvasd_resp_t);
```

Comment:
should be two lines

Reply:
I know I'm asking for many things, but these headers are a bit messy. I think there should consistently be a new if there's a free. For example, the should be a openvasd_response_new.